src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -23,14 +23,16 @@ internal static class CertificateAssetDownloader
         {
             byte[]? data = DownloadAsset(uri, ref remainingDownloadTime);
 
-            if (data == null)
+            if (data == null || data.Length == 0)
             {
                 return null;
             }
 
             try
             {
-                return new X509Certificate2(data);
+                X509Certificate2 certificate = new X509Certificate2(data);
+                certificate.ThrowIfInvalid();
```

Comment:
The `data.Length == 0` check above is enough to fix the failing test. However, if there is ever any content that does cause `X509Certificate2.ctor` to not throw, and also have an invalid handle, we should protect against those scenarios, too.